xpk.py

```diff
@@ -3243,6 +3253,55 @@ def create_cluster_if_necessary(args, gke_control_plane_version: str) -> int:
     return run_gke_cluster_create_command(args, gke_control_plane_version)
 
 
+def get_nodepool_zone(args, nodepool_name) -> tuple[int, str]:
+  """Return zone in which nodepool exists in the cluster.
+
+  Args:
+    args: user provided arguments for running the command.
+    nodepool_name: name of nodepool.
+
+  Returns:
+    Tuple of int, str where
+    int is the return code - 0 if successful, 1 otherwise.
+    str is the zone of nodepool.
+  """
+  command = (
+      f'gcloud beta container node-pools describe {nodepool_name}'
+      f' --cluster {args.cluster} --project={args.project}'
+      f' --region={zone_to_region(args.zone)} --format="value(locations)"'
+  )
+  return_code, nodepool_zone = run_command_for_value(
+      command, 'Get Node Pool Zone', args
+  )
+  if return_code != 0:
+    xpk_print(f'Get Node Pool Zone returned ERROR {return_code}')
+    return 1, None
+
+  return 0, nodepool_zone.strip()
+
+
+def check_device_type_in_cluster(args, device_type) -> bool:
+  """Check if cluster has resources of a specified device_type.
+
+  Args:
+    args: user provided arguments for running the command.
+    device_type: device_type to check.
+
+  Returns:
+    True if device_type exists in the cluster, False otherwise.
+  """
+  resources_configmap_name = f'{args.cluster}-{_CLUSTER_RESOURCES_CONFIGMAP}'
+  resources_config_map = get_cluster_configmap(args, resources_configmap_name)
+  if resources_config_map is None:
+    xpk_print(
+        f'No ConfigMap exist for cluster with the name {resources_config_map}.'
+    )
+    return False
```

Comment:
we should ignore the device check if the configmap doesn't exist to avoid breaking old clusters.

Reply:
Sure, updated!